### DISCUSSION TITLE
render components from an array with the name of the components (as strings)

### DISCUSSION THREAD

**Discussion by tryggs** [sveltejs/kit#5592] [Q&A] [answered]
Array: `["Comp1", "Comp2", "Comp3", "Comp4"]`

how would I render these four components with only the name of the component stored as strings in an array?

the solution has to be dynamic so it shouldn't matter how many components or what the component is. 
The components will always exist!

Thanks!

**Comment by stephane-vanraes** [comment, marked as answer]
You would have to create some mapping from string to actual component: 
```js
const mapping = {
   "Comp1": Comp1,
   "Comp2": Comp2
}
```
and then use that instead:
```html
<svelte:component this={mapping[component]} />
```

**Reply by tryggs** [reply]
Thank you! 
for now, this is the solution that I will go with but do you think that there is any way to pass a string as `this` directly so I don't have to create a map manually?

**Reply by stephane-vanraes** [reply]
No there isn't, `this` has to be a component.

**Reply by tryggs** [reply]
got it! Thank you
